mongo/src/main/java/com/redhat/lightblue/mongo/crud/Translator.java

```diff
@@ -200,6 +208,27 @@ public static String translateJsPath(Path p) {
     }
 
     /**
+     * Return a full field representation of the given index based on the provided entity metadata.  This will include asterisks to represent array indexes.
+     *
+     * @param md
+     * @param index
+     * @return
+     */
+    public static FieldTreeNode getFieldByIndex(EntityMetadata md, Path index) {
```

Comment:
This is wrong. It should look like this:
MutablePath p=new MutablePath()
for(i=0 to numsegments) {
  String seg=index.head(i);
  p.push(seg);
  if(md.resolve(index.prefix(i+1)) is an array)
    p.push(*)
}
return md.resolve(p);